jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java

```diff
@@ -0,0 +1,155 @@
+
+import com.github.fge.jsonpatch.JsonPatchException;
+import static io.jans.as.model.util.Util.escapeLog;
+import io.jans.as.common.model.common.User;
+import io.jans.configapi.core.rest.ProtectedApi;
+import io.jans.configapi.rest.model.SearchRequest;
+import io.jans.configapi.service.auth.UsrService;
+import io.jans.configapi.util.ApiAccessConstants;
+import io.jans.configapi.util.ApiConstants;
+import io.jans.configapi.core.util.Jackson;
+import io.jans.orm.model.PagedResult;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+
+@Path(ApiConstants.USER)
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@ApplicationScoped
+public class UserResource extends BaseResource {
+
+    private static final String USER = "user";
+
+    @Inject
+    Logger logger;
+
+    @Inject
+    UsrService userSrv;
+
+    @GET
+    @ProtectedApi(scopes = { ApiAccessConstants.USER_READ_ACCESS })
+    public Response getOpenIdConnectClients(
+            @DefaultValue(DEFAULT_LIST_SIZE) @QueryParam(value = ApiConstants.LIMIT) int limit,
+            @DefaultValue("") @QueryParam(value = ApiConstants.PATTERN) String pattern,
+            @DefaultValue(DEFAULT_LIST_START_INDEX) @QueryParam(value = ApiConstants.START_INDEX) int startIndex,
+            @QueryParam(value = ApiConstants.SORT_BY) String sortBy,
+            @QueryParam(value = ApiConstants.SORT_ORDER) String sortOrder) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("User serach param - limit:{}, pattern:{}, startIndex:{}, sortBy:{}, sortOrder:{}",
+                    escapeLog(limit), escapeLog(pattern), escapeLog(startIndex), escapeLog(sortBy),
+                    escapeLog(sortOrder));
+        }
+        SearchRequest searchReq = createSearchRequest(userSrv.getPeopleBaseDn(), pattern, sortBy, sortOrder, startIndex,
+                limit, null, null);
+
+        final List<User> users = this.doSearch(searchReq);
+        logger.debug("User serach result:{}", users);
+        return Response.ok(users).build();
+    }
+
+    @GET
+    @ProtectedApi(scopes = { ApiAccessConstants.USER_WRITE_ACCESS })
+    @Path(ApiConstants.INUM_PATH)
+    public Response getUserByInum(@PathParam(ApiConstants.INUM) @NotNull String inum) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("User serach by inum:{}", escapeLog(inum));
+        }
+        User user = userSrv.getUserByInum(inum);
+        logger.debug("user:{}", user);
+        return Response.ok(user).build();
+    }
+
+    @POST
+    @ProtectedApi(scopes = { ApiAccessConstants.USER_WRITE_ACCESS })
+    public Response createOpenIdConnect(@Valid User user) {
```

Comment:
Not sure what does method name mean `createOpenIdConnect`, sounds like typo. `createUser` fits better.

Reply:
Sorry, rectified the same